tests/mocha/xml_test.js

```diff
@@ -73,13 +76,29 @@ suite('XML', function() {
   teardown(function() {
     sharedTestTeardown.call(this);
   });
+
   suite('textToDom', function() {
     test('Basic', function() {
       const dom = Blockly.utils.xml.textToDom(this.complexXmlText);
-      chai.assert.equal(dom.nodeName, 'xml', 'XML tag');
-      chai.assert.equal(dom.getElementsByTagName('block').length, 6, 'Block tags');
+      assertXmlDoc(dom);
+      chai.assert.equal(
+          dom.getElementsByTagName('block').length, 6, 'Block tags');
+    });
+
+    test('text with NCR Control characters are properly deserialized',
+        function() {
+          const dom = Blockly.utils.xml.textToDom('<xml>&#x1;</xml>');
```

Comment:
Please add a decimal NCR as well as the hex one, since the encoder uses decimal now (if I have understood correctly).